tests/unit/expiry/TestDataProvider.unitTest.ts

```diff
@@ -364,4 +396,43 @@ describe("TestDataProvider", () => {
       });
     });
   });
+  context("for insertTestResult", () => {
+    it(`should call the 'testResultDAO.createSingle' method and 'logDefectsReporting'`, async () => {
+      testDataProvider = new TestDataProvider();
+      MockTestResultsDAO = jest.fn().mockImplementation(() => {
+        return {
+          createSingle: (something: any) => {
+            return Promise.resolve({});
+          },
+        };
+      });
+      testDataProvider.testResultsDAO = new MockTestResultsDAO();
+      const logDefectsReportingSpy = spyOn(
+        utils.LoggingUtil,
+        "logDefectsReporting"
+      );
+      expect(logDefectsReportingSpy).not.toHaveBeenCalled();
+      await testDataProvider.insertTestResult({} as models.ITestResultPayload);
+      expect(logDefectsReportingSpy).toHaveBeenCalledTimes(1);
+    });
```

Comment:
👍

Reply:
yes agree, i can disable the IDE auto save I have configured for now. We will run prettier once the last refactor ticket is in and prettier as a hook precommit.